Cookbook/Technical-Documents/BackendFeatureHowTo.md

```diff
@@ -0,0 +1,200 @@
+# How to implement a feature that works with the backend
+Let's look at a minimal and simple example of working with the backend to get some items into our view model, before moving on to more difficult topics.
+
+## A simple example
+Imagine some endpoint on the backend returns this piece of JSON:
+```
+{
+  "cities": [
+    {
+      "name": "London"
+    },
+    {
+      "name": "Bath"
+    }
+  ]
+}
+```
+
+First, we want to define some data types, to which we will map this JSON. This is known as a *response*.
+
+### Response
+A `Response` is a data type which wraps the whole response returned from a network call. In many cases it is unnecessary, just like in this example, where there is only one key. For the sake of this exercise, let's stick with it though. A later section will describe a way to extract the contents directly without creating a separate data type.
+
+A response data type is a plain struct conforming to `Equatable`  (this is usually necessary for tests) and `Decodable`. We also define an additional data type for the city itself. You will want to define all the new data types necessary to describe a response:
+```
+struct CitiesResponse: Equatable, Decodable {
+    let cities: [City]
+}
+
+struct City: Equatable, Decodable {
+    let name: String
+}
```

Comment:
I'm wondering, do we take the habit of defining these structs all at top level, or do we prefer declaring them as nested types?

```swift
struct CitiesResponse: Equatable, Decodable {
  struct City: Equatable, Decodable {
    let name: String
  }
  let cities: [City]
}
```

Reply:
AFAIK generally it makes more sense to define them at the top level, as you keep passing around the deepest-nested types around (for rendering etc.), if they were defined as subtypes you'd have to have typealiases everywhere or it would be much more typing.

Reply:
I'll make a note about that.